### PR TITLE
Exercise 7

### PR DESCRIPTION
**Pull request by steven-mi**
Pull Request for changing to nuxt.js: https://github.com/Systems-Development-and-Frameworks/abteilung6/pull/9#issue-550501741

Review for the other team: https://github.com/Systems-Development-and-Frameworks/SCP-Foundation/pull/8#pullrequestreview-576357731

### CODE REVIEW

**Comment by steven-mi** [comment]
@Systems-Development-and-Frameworks/mentors if possible, could you perhaps assess if everything is ok? Everything except the tests should be done. 

I was also not sure if it is necessary to implement delete and downvote. Could this perhaps be clarified?

File: webapp/README.md [hunk 1]
```diff
@@ -1,20 +1,26 @@
-# hackernews
```
Comment (s0557917) [inline webapp/README.md:1]: ⭐ For instructions in the README.md on how to build your webapp for production.

File: webapp/components/LoginForm/LoginForm.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { mount } from '@vue/test-utils'
```
Comment (s0557917) [inline webapp/components/LoginForm/LoginForm.spec.js:1]: ⭐ ⭐ For a login feature in your webapp including a Vue component and its software tests.

File: webapp/components/List/List.vue [hunk 3]
```diff
@@ -5,6 +5,7 @@
       :id="item.id"
       :key="item.id"
       :item="item"
+      :is-authenticated="isAuthenticated"
```
Comment (s0557917) [inline webapp/components/List/List.vue:8]: ⭐ For a delete button that is only visible to the author of the post.
❌ I was not able to find your edit button, did you implement it? Or did I oversee it?

File: webapp/components/Navbar/Navbar.spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { mount } from '@vue/test-utils'
+import Navbar from './Navbar'
+
+describe('Given a navbar', () => {
+  let wrapper
+  beforeEach(() => {
+    wrapper = mount(Navbar)
+  })
+  it('render buttons', () => {
+    expect(wrapper.text()).toContain('Login')
```
Comment (s0557917) [inline webapp/components/Navbar/Navbar.spec.js:10]: ⭐ ⭐ For a menu component which shows a login or logout button and its software tests.

File: webapp/components/ListItem/ListItem.vue [hunk 5]
```diff
@@ -1,17 +1,20 @@
 <template>
   <div id="list-item">
     <h2>{{ item.title }} - ({{ item.votes }})</h2>
-    <button @click="upvoteItem()">Upvote</button>
-    <button @click="downvoteItem()">Downvote</button>
-    <button @click="$emit('delete-item', item)">Remove</button>
+    <button v-if="isAuthenticated" @click="upvoteItem()">Upvote</button>
```
Comment (s0557917) [inline webapp/components/ListItem/ListItem.vue:4]: ⭐ For an upvote button that behaves according to the authentication state of your user

**Review by s0557917 — CHANGES_REQUESTED** [review]
⭐ ⭐ For the API connection between your front- and backend.
⭐ For no changes in "Files Changed" tab of the refactoring from vue-cli to create-nuxt-app

File: webapp/README.md [hunk 6]
```diff
@@ -1,20 +1,26 @@
...
-# generate static project
-$ npm run generate
+Build for production and launch server
+```bash
+npm run build
+npm run start
 ```
 
-For detailed explanation on how things work, check out [Nuxt.js docs](https://nuxtjs.org).
+## Deploy static project
```
Comment (Aimtracker) [inline webapp/README.md:21]: ⭐ For instructions in the README.md on how to build your webapp for production.

File: webapp/components/LoginForm/LoginForm.vue [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          Login
+        </button>
+      </label>
+    </form>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'LoginForm',
```
Comment (Aimtracker) [inline webapp/components/LoginForm/LoginForm.vue:28]: ⭐ ⭐ For a login feature in your webapp including a Vue component and its software tests.

File: webapp/components/Navbar/Navbar.vue [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    <div v-if="isAuthenticated" class="option">
+      <a id="a_logout" @click="() => $emit('logout')">Logout</a>
+    </div>
+  </nav>
+</template>
+
+<script>
+import Logo from '@/components/Logo/Logo'
+export default {
+  name: 'Navbar',
```
Comment (Aimtracker) [inline webapp/components/Navbar/Navbar.vue:24]: ⭐ ⭐ For a menu component which shows a login or logout button and its software tests.

File: webapp/components/ListItem/ListItem.vue (hunk 5 above)
Comment (Aimtracker) [inline webapp/components/ListItem/ListItem.vue:4]: ⭐ For an upvote button that behaves according to the authentication state of your user

File: webapp/components/ListItem/ListItem.vue [hunk 9]
```diff
@@ -1,17 +1,20 @@
...
+    <button v-if="isAuthenticated" @click="downvoteItem()">Downvote</button>
+    <button v-if="item.isOwner" @click="$emit('delete-item', item)">
```
Comment (Aimtracker) [inline webapp/components/ListItem/ListItem.vue:6]: ⭐ For a delete and edit button that is only visible to the author of the post.

Note: I did not find an edit button.

**Review by Aimtracker — APPROVED** [review]
I do like your concise coding style which makes it easy to read and navigate through your code. The only thing I did find was that the edit button is apparently missing, but maybe I just didn't find that? Otherwise great work!

**Review by medizinmensch — COMMENTED** [review]
Hi @abteilung6!

Thanks for the PR. 

I can entirely agree with @Aimtracker  and @s0557917 . Both noted correctly that you're missing an edit button. As it's written in the description, connection to the backend is optional. So a dummy button would have been okay. Pity!
❌    For a delete and edit button that is only visible to the author of the post.

But it looks good apart from that:

⭐ For instructions in the   README.md on how to build your webapp for production.
⭐ For   no changes in "Files Changed" tab of the refactoring from vue-cli   to create-nuxt-app. (See #1 in instructions)
⭐ ⭐ For the API connection   between your front- and backend.
⭐ For   your previous frontend tests still passing. Requests to the backend are   mocked.
⭐ ⭐ For a login feature in your   webapp including a Vue component and its software tests.
⭐ ⭐ For   a menu component which shows a login or logout button and its software tests.
⭐ For an upvote button that   behaves according to the authentication state of your user
⭐ For Lighthouse reporting that   your production website is installable as PWA (except HTTPS).
⭐ For   requesting a review and reviewing another team's PR.

But apart from that, quite nice! You could have tried to keep the PR a bit smaller. Keep that in mind for the future. 

12/13 ⭐ 




Thanks!

**Pull request merged**